statuspage/src/components/CanaryModal.vue

```diff
@@ -0,0 +1,73 @@
+<!-- This component encapsulates a modal that is displayed -->
+<!-- when clicking on canary                               -->
+<template>
+  <!--  Component modal doc: : https://bootstrap-vue.org/docs/components/modal-->
+  <!--  Here we use dynamic id for the modal since it is being called from a loop.-->
+  <!--  If ID is not distinguished then a click on any row will open modal for every value-->
+  <b-modal :id="`modal-canary${name}${namespace}`" ok-only>
+    <template #modal-title>
+      <img :src="`images/${checkType}.svg`" v-bind:style="{ height: '1.875rem' }" :alt="`${checkType} logo`"> {{name}}
+    </template>
+    <ul class="list-unstyled">
+      <li><b>Name:</b> {{name}}</li>
+      <li><b>Namespace:</b> {{namespace}}</li>
+      <li><b>Description:</b> {{description}}</li>
+      <li><b>Interval:</b> {{interval}} Seconds</li>
+      <li><b>Owner:</b> {{getProp(owner)}}</li>
+      <li><b>Severity:</b> {{getProp(severity)}}</li>
+    </ul>
+  </b-modal>
+</template>
+
+
+<script>
+export default {
+  name: "CanaryModal",
+  props: {
+    interval: {
+      type: String,
+      required: true
+    },
+    owner: {
+      type: String,
+      required: true
+    },
+    severity: {
+      type: String,
+      required: true
+    },
+    name: {
+      type: String,
+      required: true
+    },
+    namespace: {
+      type: String,
+      required: true,
+    },
+    description: {
+      type: String,
+      required: true,
+    },
+    checkType: {
+      type: String,
+      required: true,
+    },
+  },
+  methods: {
+    getProp(prop){
+      if (prop.length === 0) {
```

Comment:
```suggestion
      if (typeof prop === 'string' && prop.length === 0) {
```
Ints and bools won't have the length property.

See the typeof operator: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof